src/proof/alethe/alethe_printer.cpp

```diff
@@ -0,0 +1,321 @@
+    if (it != steps[nested_level].end())
+    {
+      Trace("alethe-printer")
+          << "... subproof is already printed " << pfn->getResult() << " "
+          << vrule << " / " << pfn->getArguments() << std::endl;
+      return prefix + "t" + std::to_string(it->second);
+    }
+
+    // If not printed before, enter next level
+    nested_level++;
+    assumptions.push_back({});
+    steps.push_back({});
+
+    // Print anchor
+    Trace("alethe-printer")
+        << "... print anchor " << pfn->getResult() << " " << vrule << " "
+        << " / " << pfn->getArguments() << std::endl;
+    out << "(anchor :step " << prefix << "t" << step_id;
+
+    // Append index of anchor to prefix so that all steps in the subproof use it
+    prefix.append("t" + std::to_string(step_id) + ".");
+    step_id++;
+
+    // If the subproof is a bind the arguments need to be printed as
+    // assignments, i.e. args=[(= v0 v1)] is printed as (:= (v0 Int) v1).
+    if (vrule == AletheRule::ANCHOR_BIND)
+    {
+      out << " :args (";
+      for (unsigned long int j = 3, size = pfn->getArguments().size(); j < size;
+           j++)
+      {
+        out << "(:= (" << pfn->getArguments()[j][0].toString() << " "
+            << pfn->getArguments()[j][0].getType().toString() << ") "
+            << pfn->getArguments()[j][1].toString() << ")";
+        if (j != pfn->getArguments().size() - 1)
+        {
+          out << " ";
+        }
+      }
+      out << ")";
+    }
+    // In all other cases there are no arguments
+    out << ")\n";
+
+    // If the subproof is a genuine subproof the arguments are printed as
+    // assumptions
+    if (vrule == AletheRule::ANCHOR_SUBPROOF)
+    {
+      for (size_t i = 3, size = pfn->getArguments().size(); i < size; i++)
+      {
+        Trace("alethe-printer")
+            << "... print assumption " << pfn->getArguments()[i] << std::endl;
+        out << "(assume " << prefix << "a" << std::to_string(i - 3) << " "
+            << pfn->getArguments()[i] << ")\n";
+        assumptions[nested_level][pfn->getArguments()[i]] = i - 3;
+      }
+    }
+
+    // Store step_id until children are printed to resume counter at current
+    // position
+    current_step_id = step_id;
+    step_id = 1;
+  }
+
+  // Assumptions are printed at the anchor and therefore have to be in the list
+  // of assumptions when an assume is reached.
+  if (vrule == AletheRule::ASSUME)
+  {
+    Trace("alethe-printer")
+        << "... reached assumption " << pfn->getResult() << " " << vrule << " "
+        << " / " << pfn->getArguments() << " " << nested_level << std::endl;
+
+    // While in most cases the assumption is printed at the same level than the
+    // step whose premise it is, it is possible that it is from a different
+    // level. Thus, the whole list needs to be traversed. Since this case is
+    // rare adapting the prefix should be rarely necessary.
+    for (int i = nested_level; i >= 0; i--)
```

Comment:
```suggestion
    for (size_t i = d_nestedLevel + 1; i > 0; i--)
```
And change all occurrences of `i` below to `i-1`.